Tasks/JenkinsQueueJobV2/job.ts

```diff
@@ -424,8 +424,17 @@ export class Job {
             } else if (httpResponse.statusCode == 404) {
                 // got here too fast, stream not yet available, try again in the future
                 thisJob.stopWork(thisJob.queue.TaskOptions.pollIntervalMillis, thisJob.State);
-            } else if (httpResponse.statusCode != 200) {
+            } else if (httpResponse.statusCode == 401) {
+                    Util.failReturnCode(httpResponse, 'Job progress tracking failed to read job progress');
+                    thisJob.queue.TaskOptions.captureConsole = false;
+                    thisJob.queue.TaskOptions.capturePipeline = false;
+                    thisJob.queue.TaskOptions.shouldFail = true;
+                    thisJob.queue.TaskOptions.failureMsg = 'Job progress tracking failed to read job progress';
+                    thisJob.stopWork(0, JobState.Finishing);
+            } 
+            else if (httpResponse.statusCode != 200) {
```

Comment:
Even though this file is using `==` in some places, it isn't type safe and I don't think it [should ever be used](https://stackoverflow.com/a/359509/3084634).  Can we fix at least this method to use use `===` and `!==`?

Reply:
I am fine with that. I usually use the !== and === forms, but was following the pattern of the file in this case. I'll update it.